components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/LightWeightKeyManagerImpl.java

```diff
@@ -0,0 +1,469 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.core.api.KeyManager;
+import org.wso2.carbon.apimgt.core.api.RestCallUtil;
+import org.wso2.carbon.apimgt.core.auth.dto.DCRClientInfo;
+import org.wso2.carbon.apimgt.core.configuration.models.KeyMgtConfigurations;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.exception.ExceptionCodes;
+import org.wso2.carbon.apimgt.core.exception.KeyManagementException;
+import org.wso2.carbon.apimgt.core.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.core.models.API;
+import org.wso2.carbon.apimgt.core.models.AccessTokenInfo;
+import org.wso2.carbon.apimgt.core.models.AccessTokenRequest;
+import org.wso2.carbon.apimgt.core.models.HttpResponse;
+import org.wso2.carbon.apimgt.core.models.KeyManagerConfiguration;
+import org.wso2.carbon.apimgt.core.models.OAuthAppRequest;
+import org.wso2.carbon.apimgt.core.models.OAuthApplicationInfo;
+import org.wso2.carbon.apimgt.core.models.Scope;
+import org.wso2.carbon.apimgt.core.util.APIMgtConstants;
+import org.wso2.carbon.apimgt.core.util.KeyManagerConstants;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * This class holds the key manager implementation for light weight key manager
+ */
+public class LightWeightKeyManagerImpl implements KeyManager {
+    private static final Logger log = LoggerFactory.getLogger(LightWeightKeyManagerImpl.class);
+    private KeyMgtConfigurations keyManagerConfigs;
+    private RestCallUtil restCallUtil;
+
+    public LightWeightKeyManagerImpl() {
+        keyManagerConfigs = ServiceReferenceHolder.getInstance().getAPIMConfiguration().getKeyManagerConfigs();
+        restCallUtil = new RestCallUtilImpl();
+    }
+
+    public LightWeightKeyManagerImpl(KeyMgtConfigurations keyManagerConfigs, RestCallUtil restCallUtil) {
+        this.keyManagerConfigs = keyManagerConfigs;
+        this.restCallUtil = restCallUtil;
+    }
+
+    @Override
+    public OAuthApplicationInfo createApplication(OAuthAppRequest oauthAppRequest) throws KeyManagementException {
+        if (log.isDebugEnabled()) {
```

Comment:
We can use https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/util/APIUtils.java#L65